### ISSUE TITLE
No save button (can't save snaps)

### ISSUE THREAD

**Issue by h4rv1n** [marzika/Snapprefs#303]
Hi. I'm currently using the #47, with Xposed and Snapchat 9.39.5.0
Problems are the Text customization (triangle on top left), which makes SC stopped working once pressed and no save button (enabled it in the snapprefs), auto save not working. Can't save anything and the unlimited viewing time not working. The only thing that works is spoofing

Please help me
Thanks,
Harvin

[xposed_error_20161103_173013.zip](https://github.com/marzika/Snapprefs/files/568560/xposed_error_20161103_173013.zip)

**Comment by M1kep** [comment]
Which build exactly are you using? Preferably a link to the Jenkins.

**Comment by h4rv1n** [comment]
Hi, thanks for your reply. 
I'm currently using build #47, since #48 doesn't work at all at my SC 9.39.5.0. 
Also, here's the link to the #47: http://138.68.21.49:8080/job/SNAPPREFS/47/

Again, thank you and have a great day :)

**Comment by jimdewhit** [comment]
I've noticed this occasionally using Dev build #33 (http://138.68.21.49:8080/job/SNAPPREFSDev/33/) with SC 9.39.5.0. 
The download button will sometimes be there, and sometimes it's missing.

I'm not sure if this is due to the type/source of the snap (I haven't paid close enough attention to that).

**Comment by h4rv1n** [comment]
@jimdewhit mine never ever appear.
I was using SC 9.31.xx with the Snapprefs 2.0.0 beta from Xposed repo, until yesterday, snapchat "has stopped working" and force me to update. Everything was very fine.

**Comment by jimdewhit** [comment]
@h4rv1n I was in the same boat. 

I installed Dev build 33 (referenced by @M1kep in Issue #302), and it (mostly) works great. There ARE still bugs. I get random crashes every once in a while and chat seems a bit sluggish, but it's a development build so I'm not complaining. Too much... 

I figured I would add my voice to this issue, as it's one that I've come across in the build I'm using as well.

**Comment by h4rv1n** [comment]
@jimdewhit Glad to hear that I'm not alone 
Thanks for sharing

**Comment by M1kep** [comment]
@h4rv1n your problem is that you are still using the 9.31 version. 

Give http://138.68.21.49:8080/job/SNAPPREFSDev/lastStableBuild/ a try :)

@jimdewhit if you could create an issue with your crash and logs that but much appreciated!

**Comment by h4rv1n** [comment]
@M1kep I'm using 9.39.5 version of sc.

**Comment by h4rv1n** [comment]
@M1kep I'm totally sure

**Comment by jimdewhit** [comment]
@h4rv1n I think @M1kep means that the version of Snapprefs that you're using is meant for SC 9.31.x. Try installing the latest Snapprefs Dev build.

@M1kep I'll keep an eye out for the random crashes and create an issue with the logs when I catch one.

**Comment by h4rv1n** [comment]
@jimdewhit I'm using second from the latest version of dev build. The latest, which is #48, doesn't work at all for the snapchat, the version that I used (#47) is for 9.39.5.0; written on the changelog.

**Comment by M1kep** [comment]
@h4rv1n again. That branch you linked is not the dev branch. The very bottom of that page you linked shows the branch that was built from, orgin/stabletey to use the one I linked. http://138.68.21.49:8080/job/SNAPPREFSDev/lastStableBuild/

**Comment by h4rv1n** [comment]
Thank you @M1kep 
What was the one that I used for (the #47)? 
I didnt get it 
I'll give a try

**Comment by h4rv1n** [comment]
@M1kep Hi!!! It works perfectly! Thank you very much!!!
Can you guide me like telling me the difference between those 2 (Snapprefs that I downloaded wrongly and SnapprefsDev), also, for the next update, I'll just need to go to the SnapprefsDev Jenkins, right?

**Comment by M1kep** [comment]
@h4rv1n No worries! So currently we are developing the 9.39.5 version so it is on the Development branch which is built on this Jenkins URL: http://138.68.21.49:8080/job/SNAPPREFSDev/lastStableBuild/

The "stable" branch is still set to 9.31 because we are working out some final things on the development build before pushing it out to the stable branch, which is built on this Jenkins URL: http://138.68.21.49:8080/job/SNAPPREFS/lastStableBuild/

In regards to the commit message on your first few posts, I have no clue how that ended up there, and might have broken chat saving for stable.

In conclusion her is the comparison of the two lings:

"Stable" -> 9.31: http://138.68.21.49:8080/job/SNAPPREFS/lastStableBuild/
"Development" -> 9.39.5: http://138.68.21.49:8080/job/SNAPPREFSDev/lastStableBuild/

Depending on when you update again you may want to use the "Stable" download if we have moved dev branch to the stable branch, when that happens I may disable

I will look into possibly getting a currently supported version listed on the readme.

**Comment by h4rv1n** [comment]
Thanks for the explanation @M1kep 
I'll close the issue now. 
Have a nice day, thanks again!!! :)
